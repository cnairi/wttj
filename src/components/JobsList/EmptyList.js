import React from 'react'
import { Alert } from '@welcome-ui/alert'

import { EMPTY_JOBS_LIST_TEXT, EMPTY_JOBS_LIST_TITLE } from '../../constants'

function EmptyList() {
  return (
    <Alert variant="info">
      <Alert.Title>{EMPTY_JOBS_LIST_TITLE}</Alert.Title>
      <span>{EMPTY_JOBS_LIST_TEXT}</span>
    </Alert>
  )
}

export default EmptyList
