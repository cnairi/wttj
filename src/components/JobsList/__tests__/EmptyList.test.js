import React from 'react'
import EmptyList from '@components/JobsList/Content/Empty'
import { render } from '@utils/tests'

jest.mock('react-i18next', () => ({
  useTranslation: () => {
    return {
      t: str => str,
      i18n: {
        changeLanguage: () => new Promise(() => {}),
      },
    }
  },
}))

describe('Component JobsList <EmptyList />', () => {
  test('It should renders properly', () => {
    const tree = render(<EmptyList />)
    expect(tree).toMatchSnapshot()
  })

  test('It have title and text', async () => {
    const { getByRole, getByText } = render(<EmptyList />)
    const titleElement = getByRole('heading')
    const textElement = getByText('filters.empty_list_title')
    expect(titleElement).toHaveTextContent('filters.empty_list_title')
    expect(textElement).toBeDefined()
  })
})
