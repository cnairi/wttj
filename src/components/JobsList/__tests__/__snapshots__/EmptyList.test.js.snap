// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Common <Language /> It should renders properly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c1 {
  font-family: welcome-font,sans-serif;
  font-weight: 600;
  font-size: 1rem;
  line-height: 1.2;
  display: block;
}

.c4 {
  width: 0.9375rem;
  height: 0.9375rem;
}

.c4 g,
.c4 path {
  stroke: none;
}

.c3 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  margin-right: 0.375rem;
  color: #1B57B2;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c3.c3 {
  margin-right: 0.5rem;
}

.c0 {
  font-size: 0.875rem;
  border-width: 1px;
  border-style: solid;
  border-radius: 6px;
  padding: 1.25rem;
  color: #4C4C4C;
  background-color: #EAF2FE;
  border-color: #1B57B2;
  padding-top: 1.25rem;
  padding-bottom: 1.25rem;
}

.c0.c0 {
  padding-top: 1.25rem;
  padding-bottom: 1.25rem;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0;
  margin-bottom: 0.625rem;
  color: #1B57B2;
}

.c2:only-child {
  margin-bottom: 0;
}

<body>
    <div>
      <div
        class="c0"
        variant="info"
      >
        <h5
          class="c1 c2 wui-text"
        >
          <div
            class="c3"
            variant="info"
          >
            <svg
              alt="Information"
              class="c4"
              role="img"
              viewBox="0 0 100 100"
            >
              <path
                d="M49.66 93.76c24.456-2.769 44.353-24.788 44.353-49.084 0-11.56-4.496-21.915-12.66-29.157-7.049-6.252-16.119-9.567-25.946-9.567-1.67 0-3.36.096-5.07.29-24.453 2.767-44.35 24.786-44.35 49.08 0 11.562 4.496 21.92 12.66 29.16 8.245 7.313 19.266 10.61 31.013 9.277zm35.678-82.68C94.793 19.464 100 31.397 100 44.676c0 27.225-22.28 51.895-49.662 54.996-1.94.22-3.866.328-5.768.328-11.29 0-21.748-3.839-29.91-11.08C5.207 80.535 0 68.604 0 55.323 0 28.1 22.278 3.43 49.66.328 63.14-1.196 75.804 2.622 85.338 11.08zM43.71 31.12a7.784 7.784 0 0 1-.342-1.122 6.458 6.458 0 0 1-.103-1.156c0-.391.034-.782.103-1.156.085-.391.205-.765.342-1.122a5.95 5.95 0 0 1 .564-1.02 5.29 5.29 0 0 1 .736-.901 5.4 5.4 0 0 1 .907-.749 6.079 6.079 0 0 1 1.043-.544c.36-.153.736-.272 1.112-.34a6.04 6.04 0 0 1 4.5.884c.324.221.632.46.906.749.273.272.532.578.752.9.207.324.394.664.548 1.02.137.358.256.732.342 1.123a6.497 6.497 0 0 1 0 2.312 7.784 7.784 0 0 1-.342 1.122c-.154.357-.34.697-.548 1.02a7.52 7.52 0 0 1-.752.918 6.3 6.3 0 0 1-.907.73 4.953 4.953 0 0 1-1.026.545 5.708 5.708 0 0 1-1.13.34 5.517 5.517 0 0 1-1.163.119c-.393 0-.786-.034-1.18-.12a5.239 5.239 0 0 1-1.112-.34 5.079 5.079 0 0 1-1.043-.543 6.3 6.3 0 0 1-.907-.731 6.192 6.192 0 0 1-.736-.918 5.95 5.95 0 0 1-.564-1.02zm13.64 36.81c2.82-.32 5.36 1.692 5.68 4.49.321 2.798-1.701 5.325-4.517 5.644a12.99 12.99 0 0 1-10.076-3.18 12.831 12.831 0 0 1-4.32-9.589V52.548l-3.023.343c-2.828.317-5.362-1.691-5.68-4.49-.321-2.8 1.702-5.326 4.516-5.645l8.737-.99a5.155 5.155 0 0 1 4 1.263 5.088 5.088 0 0 1 1.713 3.805v18.461c0 1.034.558 1.683.892 1.979a2.65 2.65 0 0 0 2.078.656z"
                fill="currentColor"
                fill-rule="evenodd"
              />
            </svg>
          </div>
          filters.empty_list_title
        </h5>
        <span>
          filters.empty_list_text
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-AykKC sc-LzLsE sc-LzLtg dRDxjA"
      variant="info"
    >
      <h5
        class="sc-LzLsG iiKYiV sc-LzLth iascGB wui-text"
      >
        <div
          class="sc-AykKC sc-LzLtf dBugsk"
          variant="info"
        >
          <svg
            alt="Information"
            class="sc-LzLte goFMPg"
            role="img"
            viewBox="0 0 100 100"
          >
            <path
              d="M49.66 93.76c24.456-2.769 44.353-24.788 44.353-49.084 0-11.56-4.496-21.915-12.66-29.157-7.049-6.252-16.119-9.567-25.946-9.567-1.67 0-3.36.096-5.07.29-24.453 2.767-44.35 24.786-44.35 49.08 0 11.562 4.496 21.92 12.66 29.16 8.245 7.313 19.266 10.61 31.013 9.277zm35.678-82.68C94.793 19.464 100 31.397 100 44.676c0 27.225-22.28 51.895-49.662 54.996-1.94.22-3.866.328-5.768.328-11.29 0-21.748-3.839-29.91-11.08C5.207 80.535 0 68.604 0 55.323 0 28.1 22.278 3.43 49.66.328 63.14-1.196 75.804 2.622 85.338 11.08zM43.71 31.12a7.784 7.784 0 0 1-.342-1.122 6.458 6.458 0 0 1-.103-1.156c0-.391.034-.782.103-1.156.085-.391.205-.765.342-1.122a5.95 5.95 0 0 1 .564-1.02 5.29 5.29 0 0 1 .736-.901 5.4 5.4 0 0 1 .907-.749 6.079 6.079 0 0 1 1.043-.544c.36-.153.736-.272 1.112-.34a6.04 6.04 0 0 1 4.5.884c.324.221.632.46.906.749.273.272.532.578.752.9.207.324.394.664.548 1.02.137.358.256.732.342 1.123a6.497 6.497 0 0 1 0 2.312 7.784 7.784 0 0 1-.342 1.122c-.154.357-.34.697-.548 1.02a7.52 7.52 0 0 1-.752.918 6.3 6.3 0 0 1-.907.73 4.953 4.953 0 0 1-1.026.545 5.708 5.708 0 0 1-1.13.34 5.517 5.517 0 0 1-1.163.119c-.393 0-.786-.034-1.18-.12a5.239 5.239 0 0 1-1.112-.34 5.079 5.079 0 0 1-1.043-.543 6.3 6.3 0 0 1-.907-.731 6.192 6.192 0 0 1-.736-.918 5.95 5.95 0 0 1-.564-1.02zm13.64 36.81c2.82-.32 5.36 1.692 5.68 4.49.321 2.798-1.701 5.325-4.517 5.644a12.99 12.99 0 0 1-10.076-3.18 12.831 12.831 0 0 1-4.32-9.589V52.548l-3.023.343c-2.828.317-5.362-1.691-5.68-4.49-.321-2.8 1.702-5.326 4.516-5.645l8.737-.99a5.155 5.155 0 0 1 4 1.263 5.088 5.088 0 0 1 1.713 3.805v18.461c0 1.034.558 1.683.892 1.979a2.65 2.65 0 0 0 2.078.656z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
        </div>
        filters.empty_list_title
      </h5>
      <span>
        filters.empty_list_text
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
